app/src/store/user.mock.ts

```diff
@@ -0,0 +1,153 @@
+        return { status: 'rejected', reason: 'CHECK_LOGIN_STATUS' };
+      });
+    builder
+      .addCase(login.pending, () => {
+        return { status: 'pending' };
+      })
+      .addCase(login.fulfilled, (_, action) => {
+        return { status: 'fulfilled', ...action.payload };
+      })
+      .addCase(login.rejected, (_, action) => {
+        return {
+          status: 'rejected',
+          reason: action.payload?.reason,
+          error: action.payload?.error || action.error,
+        };
+      });
+    builder
+      .addCase(logout.pending, () => {
+        return { status: 'pending' };
+      })
+      .addCase(logout.fulfilled, (_, action) => {
+        return { status: 'fulfilled', ...action.payload };
+      })
+      .addCase(logout.rejected, (_, action) => {
+        return {
+          status: 'rejected',
+          reason: action.payload?.reason,
+          error: action.payload?.error || action.error,
+        };
+      });
+  },
+});
+
+export const checkLoginStatus = createAsyncThunk<
+  {
+    accessToken: string;
+    email: string;
+  },
+  void,
+  {
+    state: State;
+    dispatch: Dispatch;
+    rejectValue: undefined;
+  }
+>('user/checkLoginStatus', async () => {
+  return { accessToken: '', email: '' };
+});
+
+export const login = createAsyncThunk<
+  { accessToken: string; email: string },
+  void,
+  {
+    state: State;
+    dispatch: Dispatch;
+    rejectValue: {
+      reason:
+        | 'FETCH_ERROR'
+        | 'AUTH0_TOKEN_RESPONSE_NOT_OK'
+        | 'FETCH_JSON_ERROR'
+        | 'LOGIN_ERROR';
+      error?: any;
+    };
+  }
+>('user/login', async (_, { rejectWithValue }) => {
+  try {
+    const fetchResult = await settlePromise(
+      fetch(`${defaultUrl}/auth0`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }),
+    );
+
+    if (fetchResult.status === 'rejected') {
+      return rejectWithValue({
+        reason: 'FETCH_ERROR',
+        error: fetchResult.reason,
+      });
+    }
+
+    const auth0TokenResponse = fetchResult.value;
+    if (!auth0TokenResponse.ok) {
+      return rejectWithValue({
+        reason: 'AUTH0_TOKEN_RESPONSE_NOT_OK',
+      });
+    }
+
+    const auth0TokenResult = await settlePromise(fetchResult.value.json());
+    if (auth0TokenResult.status === 'rejected') {
+      return rejectWithValue({
+        reason: 'FETCH_JSON_ERROR',
+        error: auth0TokenResult.reason,
+      });
+    }
+
+    return {
+      accessToken: auth0TokenResult.value.access_token,
+      email: 'test_email@twilio.com',
+    };
+  } catch (error) {
+    return rejectWithValue({ reason: 'LOGIN_ERROR', error });
+  }
+});
+
+export const logout = createAsyncThunk<
+  { accessToken: string; email: string },
+  void,
+  {
+    state: State;
+    dispatch: Dispatch;
+    rejectValue: { reason: 'LOGOUT_ERROR'; error: any };
+  }
+>('user/logout', async () => {
+  return { accessToken: '', email: '' };
+});
```

Comment:
What is the difference between user.ts and user.mock.ts? What is its purpose and when is it used?

Reply:
Ah, it seems like this mock file calls /auth0 route. I'm assuming there are duplicate code here right? We only change the part where we call auth0 API to get an auth0 token?

Reply:
Yes `user.mock.ts` is a replacement for `user.ts` when we run the app in `mocked` mode

https://wix.github.io/Detox/docs/guide/mocking/

Reply:
Is it possible to minimize what we mock here? For example, if we need to change user.ts, we also need to update the mock file and make sure they are in sync. Is it possible to move the block of code related to auth0 only to its own file that way we only mock very minimal code?

Reply:
What it does is it replaces the whole file. To mock only minimal code, I could split up the user.ts file into different files and only mock those files. However that could get messy since we will have to split every function into its own file. What do you think?

Reply:
> Is it possible to minimize what we mock...
Plus one from me on this. One idea is to move all of the auth0 stuff into something like `util/auth.ts`

Reply:
okay refactored by putting the `thunks` into a `util/auth` file. Then just mocked the `auth` file. I agree it looks cleaner this way